Yosemite/Yosemite/Model/Payments/CardPresentPaymentsConfiguration.swift

```diff
@@ -1,64 +1,83 @@
 import Foundation
 
 public struct CardPresentPaymentsConfiguration {
+    private let countryCode: String
+    private let stripeTerminalforCanadaEnabled: Bool
```

Comment:
Maybe I'm getting picky, but I'd rather revert to the earlier approach of having `purchaseCardReaderUrl` be a stored property that is calculated during initialization, rather than storing a `stripeTerminalforCanadaEnabled` flag in every possible configuration, even if it's for another country.

🤔 I guess this still needs to depend on the plugin so it can't be completely static, but my idea with the configuration was that it would be somewhat static and keep the logic somewhere else.

I made an attempt at storing a function that would calculate the URL based on plugin as a property, but I think it got a bit convoluted 😬 https://gist.github.com/koke/fdf4ee5c4249ef16a588c2e84d20770c

While not ideal, what do you think of having two separate properties for WCPay and Stripe URLs? If Stripe is going to use the same URL on every scenario, then it seems like a simple solution, and we can keep the existing implementation for WCPay.Then the caller can decide which property to use depending on the active plugin without adding all that logic to the configuration.

Reply:
I thought about that too, but looking ahead to the learn more link, that means all those callers will have to replicate that if stripe x else y logic. Having the logic in one place in the configuration struct makes more sense to me.

I'll noodle on this a bit more

Reply:
Hey @koke - how about we proceed with this and then refactor after you see how the learn-more URLs work.

Down the road I wouldn't be surprised if we add support for in-person payments for more than just WCPay and Stripe (i.e. other payment gateways). We can keep configuration as dumb/static as possible still, but allowing it to tailor which of its static responses it provides based on country and plugin/account.

Reply:
Sounds right. I'll keep thinking about this since it feels like there might be two distinct configuration layers: one more static that contains what's available according to country and feature flags, and other closer to "state", which depends on the store's existing plugin/account setup.